### PR TITLE
Chore: refactor to reduce code complexity

### PR DESCRIPTION
**Pull request by Wylwi**
This PR is simply refactoring to reduce code complexity on functions flagged by Sonar

### CODE REVIEW

File: app/src/main/java/ch/hikemate/app/ui/map/HikeDetailsScreen.kt [hunk 1]
```diff
@@ -583,6 +575,28 @@ fun DateDetailRow(
   }
 }
 
+@OptIn(ExperimentalMaterial3Api::class)
```
Comment (davidholzwarth) [inline app/src/main/java/ch/hikemate/app/ui/map/HikeDetailsScreen.kt:578]: Nitpick: Please add a docstring

File: app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt [hunk 2]
```diff
@@ -510,6 +474,58 @@ fun MapScreen(
       }
 }
 
+@Composable
+private fun LaunchedEffectForHikeUpdate(
```
Comment (davidholzwarth) [inline app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt:478]: Nitpick: Please add a docstring

**Review by davidholzwarth — CHANGES_REQUESTED** [review]
Please add some docstrings

File: app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt (hunk 2 above)
Comment (Wylwi) [reply app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt:478]: Done!

File: app/src/main/java/ch/hikemate/app/ui/map/HikeDetailsScreen.kt (hunk 1 above)
Comment (Wylwi) [reply app/src/main/java/ch/hikemate/app/ui/map/HikeDetailsScreen.kt:578]: Done!

**Review by davidholzwarth — APPROVED** [review]
LGTM

**Pull request merged**